### ISSUE TITLE
As a CIF User, I want to see and update a Project Summary Report page for an IA stream project

### ISSUE THREAD

**Issue by nanyangpro** [bcgov/cas-cif#1061]
#### Description:

Design: https://www.figma.com/file/PzOY8RyQnVXnjZJdSv2AqQ/Wireframing_Figma?node-id=4165%3A2891

Instead of an Emissions Intensity Report page for an EP stream project, an IA stream project should display a Project Summary Report page. Conditional form display will be handled in other tickets, so this card is purely about creating the Project Summary Report form.

This form is a combination of the tables `reporting_requirement` and `payment`. We can use a custom commit handler like the one currently used for milestones to insert the data from a single form_schema into the correct tables.

#### Acceptance Criteria:

In create/edit mode:

Given a CIF user has started or is editing an IA stream project
When they view the tasklist
They see an item for `Project Summary Report` (including status and "add/edit" text as appropriate)

Given a CIF user has navigated to the `Project Summary Report` form
When they fill out the form
They can fill out all the fields shown in the wireframes

Given a CIF user has filled out `Project Summary Report` form
When they click submit
They see validation errors if they missed a required field

Given a CIF user has filled out `Project Summary Report` form
When they click submit
The form submits if there are no validation errors

Given a CIF user arrives at the `Review and Submit` page
When they view the summary of changes
They see a section for `Project Summary Report` with diffs

In view mode:

Given a CIF user has created an IA stream project
When they navigate to the existing project and view the tasklist
They see an item for `Project Summary Report`

Given a CIF user clicks on the `Project Summary Report` in the tasklist
When they are taken to the form
They see the read-only information

#### Development Checklist:

- [x] new rjsf json schema in the `schema/data/prod/json_schema` directory(reuse the reporting requirement one for the first part of the form)
- [x] add a custom commit handler similar to `cif_private.handle_milestone_form_change_commit()`, named `cif_private.handle_project_summary_report_form_change_commit()`
- [x] add `/data/insert_json_schema_form_data_project_summary.sql` to insert new json-schema form data into cif.form and use custom commit handler
- [x] add "Project Summary Report" to `cif.report_type`
- [x] add Project Summary Report mutations to `app/mutations/ProjectSummaryReport`
- [ ] add a new entry to `schema/data/prod/form_001` to insert the newly created json_schema ??
- [x] update the `create_project_revision` SQL custom mutation to include the form_change record for the `project_summary_report` when iterating over all related form_change records for the revision
- [x]  add `mutations/ProjectSummaryReport` mutations
- [x] a ProjectSummaryReportForm.tsx form exists to display and edit the data as per the wireframe
- [x] a ProjectSummaryReportFormSummary.tsx exists to show the summary of the entered data
- [x] update the IA formStructure to include the new form in the IA tasklist
- [x] update the `Review and Submit` page to only show summaries related to the project funding stream
- [ ] Add tests for the IA flow (e2e, integration, and pgTap)
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by dleard** [comment]
This work is not necessarily blocked by #1051, but we will not be able to see this in the task list until #1051 is done

**Comment by Dianadec** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/1061?planning-poker) with Zenhub @BCerki @dleard @gurjmatharu @mikevespi @pbastia @sam-warren @Sepehr-Sobhani @tmastrom

**Comment by BCerki** [comment]
@nanyangpro , @suhafa , there's a question in the AC about validation: They see validation errors if they missed a required field (any other validation needed?)

**Comment by Dianadec** [comment]
From Sprint planning: Backend updates need to be added (related to amendments). cc @tmastrom

**Comment by LindsayMacfarlane** [comment]
Thank you @tmastrom ! This was certainly a complex ticket to get through. I appreciate all your effort on this one. I tested in dev by following the acceptance criteria. Everything as outlined was included and functional :)

In create/edit mode:

Given a CIF user has started or is editing an IA stream project
When they view the tasklist
They see an item for Project Summary Report (including status and "add/edit" text as appropriate)
####DONE

Given a CIF user has navigated to the Project Summary Report form
When they fill out the form
They can fill out all the fields shown in the wireframes
#### DONE

Given a CIF user has filled out Project Summary Report form
When they click submit
They see validation errors if they missed a required field
#### DONE

Given a CIF user has filled out Project Summary Report form
When they click submit
The form submits if there are no validation errors
#### DONE

Given a CIF user arrives at the Review and Submit page
When they view the summary of changes
They see a section for Project Summary Report with diffs
#### DONE

In view mode:

Given a CIF user has created an IA stream project
When they navigate to the existing project and view the tasklist
They see an item for Project Summary Report
#### DONE

Given a CIF user clicks on the Project Summary Report in the tasklist
When they are taken to the form
They see the read-only information
#### DONE
